include/nuttx/list.h

```diff
@@ -299,102 +311,6 @@ list_remove_tail(FAR struct list_node *list)
     }
 }
 
-static inline FAR struct list_node *
-list_peek_head(FAR struct list_node *list)
-{
-  if (list->next != list)
-    {
-      return list->next;
-    }
-  else
-    {
-      return NULL;
-    }
-}
-
-static inline FAR struct list_node *
-list_peek_tail(FAR struct list_node *list)
-{
-  if (list->prev != list)
-    {
-      return list->prev;
-    }
-  else
-    {
-      return NULL;
-    }
-}
-
-static inline FAR struct list_node *list_prev(FAR struct list_node *list,
-                                              FAR struct list_node *item)
-{
-  if (item->prev != list)
-    {
-      return item->prev;
-    }
-  else
-    {
-      return NULL;
-    }
-}
-
-static inline FAR struct list_node *
-list_prev_wrap(FAR struct list_node *list, FAR struct list_node *item)
-{
-  if (item->prev != list)
-    {
-      return item->prev;
-    }
-  else if (item->prev->prev != list)
-    {
-      return item->prev->prev;
-    }
-  else
-    {
-      return NULL;
-    }
-}
-
-static inline FAR struct list_node *list_next(FAR struct list_node *list,
-                                              FAR struct list_node *item)
-{
-  if (item->next != list)
-    {
-      return item->next;
-    }
-  else
-    {
-      return NULL;
-    }
-}
-
-static inline FAR struct list_node *
-list_next_wrap(FAR struct list_node *list, FAR struct list_node *item)
-{
-  if (item->next != list)
-    {
-      return item->next;
-    }
-  else if (item->next->next != list)
-    {
-      return item->next->next;
-    }
-  else
-    {
-      return NULL;
-    }
-}
-
-static inline bool list_is_empty(FAR struct list_node *list)
-{
-  return list->next == list;
-}
-
-static inline bool list_is_clear(FAR struct list_node *list)
-{
-  return list->next == NULL;
-}
-
 static inline size_t list_length(FAR struct list_node *list)
```

Comment:
So this will be changed in next P?

Reply:
It's hard or impossible if we can't use some compiler extension(e.g. https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html).